blockchain/types/tx_internal_data_cancel.go

```diff
@@ -191,8 +190,8 @@ func (t *TxInternalDataCancel) SetSignature(s TxSignatures) {
 	t.TxSignatures = s
 }
 
-func (t *TxInternalDataCancel) IntrinsicGas(currentBlockNumber uint64) (uint64, uint64, error) {
-	return params.TxGasCancel, 0, nil
+func (t *TxInternalDataCancel) IntrinsicGas(currentBlockNumber uint64) (uint64, error) {
+	return GetTxGasForTxType(t.Type())
```

Comment:
How about using the same method in each `IntrisicGas`?

```suggestion
	return GetTxGasForTxTypeWithAccountKey(t.Type(), nil, currentBlockNumber, false)
```

Reply:
Done, thanks for the suggestion.